tests/e2e/performance/test_pvc_multi_clone_performance.py

```diff
@@ -0,0 +1,409 @@
+import datetime
+import logging
+import os
+import subprocess
+import tempfile
+import time
+
+import yaml
+
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version,
+    skipif_ocp_version,
+    E2ETest,
+    performance,
+)
+from ocs_ci.ocs import constants, exceptions
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.ocs.resources import pod
+from ocs_ci.utility.utils import ocsci_log_path
+
+log = logging.getLogger(__name__)
+
+
+@performance
+@skipif_ocp_version("<4.6")
+@skipif_ocs_version("<4.6")
+class TestPvcMultiClonePerformance(E2ETest):
+    """
+    Tests to measure PVC clones creation performance ( time and speed)
+    The test is supposed to create the maximum number of clones for one PVC
+    """
+
+    def test_pvc_multiple_clone_performance(
+        self,
+        interface_iterate,
+        teardown_factory,
+        storageclass_factory,
+        pvc_factory,
+        pod_factory,
+    ):
+        """
+        1. Creating PVC
+           PVS size is calculated in the test and depends on the storage capacity, but not less then 1 GiB
+           it will use ~75% capacity of the Storage, Min storage capacity 1 TiB
+        2. Fill the PVC with 70% of data
+        3. Take a clone of the PVC and measure time and speed of creation by reading start creation and end creation
+            times from relevant logs
+        4. Repeat the previous step number of times (maximal num_of_clones is 512)
+        5. Rrint all measured statistics for all the clones.
+
+        Raises:
+            StorageNotSufficientException: in case of not enough capacity on the cluster
+
+        """
+        num_of_clones = 512
+        if interface_iterate == constants.CEPHBLOCKPOOL:
+            num_of_clones = 512  # bz_1896831
+
+        # Getting the total Storage capacity
+        ceph_cluster = CephCluster()
+        ceph_capacity = int(ceph_cluster.get_ceph_capacity())
+
+        # Use 70% of the storage capacity in the test
+        capacity_to_use = int(ceph_capacity * 0.7)
+
+        # since we do not want to use more then 65%, we add 35% to the needed
+        # capacity, and minimum PVC size is 1 GiB
+        need_capacity = int((num_of_clones + 2) * 1.35)
+        # Test will run only on system with enough capacity
+        if capacity_to_use < need_capacity:
+            err_msg = (
```

Comment:
I am not sure how accurate the math should be here, but it seems like certain borderline values slightly above 65% would pass this check.  I think that the int rounding introduces some slight errors here.  For example, a ceph_capacity of 199 and a num_of_clones value of 101 should fail this check but I think passes this check (correct me if my math is wrong).

Reply:
@wusui - You are right in your math! However, since we are doing here approximate values of capacity, it's basically my decision when to fail the test or let the clones actually be created, and I perefer to leave it this way.

Reply:
Okay.  Seems to be close to 65%, give or take a little.  I guess that this is not hard and fast, rather "it's bad if it gets too much over 65%"